pageserver/src/walreceiver.rs

```diff
@@ -172,3 +174,13 @@ impl<E: Clone> TaskHandle<E> {
         }
     }
 }
+
+// XXX: do not store this Arc anywhere, rather treat is as a MutexGuard.
+fn acquire_timeline_read(
```

Comment:
Remove the XXX comment and turn it into a doc comment.

Regarding the name, I guess your intention is to make it appear like a MutexGuard.

But at the callsite, you don't know that at a quick glance.
IMO, it's better to just call it what it is, e.g., `try_upgrade_timeline_arc`

Reply:
Moved this a `TimelineGuard`, since the pattern started to get repetitive.
Commented it there.